src/redux/sagas/messages.ts

```diff
@@ -1,24 +1,44 @@
-import { fork, take, cancel } from 'redux-saga/effects';
+import { fork, take, takeEvery, cancel, call, put } from 'redux-saga/effects';
+import { buffers } from 'redux-saga';
 import { reduxSagaFirebase } from '../../gateways/firebase';
 
 import { MessageActionType, MessageAction } from '../actions/messages';
 
-function * syncMessagesSaga () {
-  // todo: sync messages for the current channel
-  yield fork(
-    reduxSagaFirebase.database.sync,
-    'messages',
-    {
-      successActionCreator: MessageAction.syncMessages,
-    },
-    'value'
+
+// watch messages on a given address
+
+// ? send messages to a given address
+
+function * sendMessage(action: any) {
```

Comment:
I suppose we could make a `sendMessageAction` interface just above, specifying the `to` and `data` members?

Reply:
I'm confused about how best to do this. We _should_ already have the type from the action, but I don't seem to be able to access it (as in `MessageAction.sendMessage` isn't actually a type). I think this is because we bundle all the types together into `MessageAction` but can't access them separately. We probably need to alter how we do the types of actions to fix this ...

Reply:
As in, for the proper fix.